### PR TITLE
Improve fix-all progress bar experience

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**
https://github.com/dotnet/roslyn/assets/4564579/08cbe773-defc-4467-b6cf-1d6b0cb31e97

### CODE REVIEW

File: src/EditorFeatures/Core/CodeActions/IUIThreadOperationContextExtensions.cs [hunk 1]
```diff
@@ -27,14 +27,21 @@ public void Report(CodeAnalysisProgress value)
...
 
-            if (value.IncompleteItemsValue != null)
-                Interlocked.Add(ref _totalItems, value.IncompleteItemsValue.Value);
-
-            if (value.CompleteItemValue != null)
-                Interlocked.Add(ref _completedItems, value.CompleteItemValue.Value);
+            if (value.ClearValue)
+            {
+                Interlocked.Exchange(ref _totalItems, 0);
+                Interlocked.Exchange(ref _completedItems, 0);
```
Comment (CyrusNajmabadi) [inline src/EditorFeatures/Core/CodeActions/IUIThreadOperationContextExtensions.cs:33]: whoops.  we just lost this in a prior refactoring in this space.

File: src/Workspaces/Core/Portable/Workspace/Workspace.cs [hunk 2]
```diff
@@ -1466,6 +1467,7 @@ internal virtual bool TryApplyChanges(Solution newSolution, IProgress<CodeAnalys
 
             foreach (var projectChanges in projectChangesList)
             {
+                progressTracker.Report(CodeAnalysisProgress.Description(string.Format(WorkspacesResources.Applying_changes_to_0, projectChanges.NewProject.Name)));
```
Comment (CyrusNajmabadi) [inline src/Workspaces/Core/Portable/Workspace/Workspace.cs:1470]: now, when we're doing the *actual* application of hte changes to the workspace, you now see progress for that as well.

File: src/EditorFeatures/Core.Wpf/Suggestions/SuggestedActions/SuggestedAction.cs [hunk 3]
```diff
@@ -165,6 +165,7 @@ private async Task InvokeWorkerAsync(IProgress<CodeAnalysisProgress> progressTra
                 // Clear the progress we showed while computing the action.
                 // We'll now show progress as we apply the action.
                 progressTracker.Report(CodeAnalysisProgress.Clear());
+                progressTracker.Report(CodeAnalysisProgress.Description(EditorFeaturesResources.Applying_changes));
```
Comment (CyrusNajmabadi) [inline src/EditorFeatures/Core.Wpf/Suggestions/SuggestedActions/SuggestedAction.cs:168]: Now, after cleanup, and selecting "apply" we actually "Applying changes" :)

So user sees 3 passes.

First: Gettins the initial fixes.
Second: Cleaning them. 
Third: Applying them to workspace.

File: src/Workspaces/Core/Portable/Workspace/Workspace.cs [hunk 4]
```diff
@@ -11,6 +11,7 @@
 using System.Threading;
 using System.Threading.Tasks;
 using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CodeActions;
```
Comment (ToddGrun) [inline src/Workspaces/Core/Portable/Workspace/Workspace.cs:14]: nit: needed?

**Review by ToddGrun — APPROVED** [review]
:shipit:

**Pull request merged**